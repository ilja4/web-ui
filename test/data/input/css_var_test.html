<!DOCTYPE html>
<!--
Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
for details. All rights reserved. Use of this source code is governed by a
BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <link rel="import" href="css_var_comp1.html">
  <link rel="stylesheet" href="css_import.css">
  <link rel="stylesheet" href="resources/assets/css/asset.css">
  <script src="packages/web_ui/testing/testing.js"></script>
  <title>CSS Import Test</title>
</head>
<body>
  <style>
    #box-1 {
      padding: 10px;
      border: 1px solid black;
    }
    /*
      TODO(terry): .comp-1 is used in component even if the attribute
                   apply-author-styles not specified?  How is this really
                   different then reset-style-inherifance.
     */
    .comp-1 {
      font: bold 16pt arial;
    }
    .comp-1-input {
      color: blue;
      background-color: GreenYellow;
    }
    .comp-1-done {
      font: 32pt bold;
      color: DarkViolet;
      background-color: LightCyan;
    }

    /* TODO(jmesserly): workaround to make the render tree cross-platform. */
    input {
      padding: 0px;
      border: 0px;
      width: 300px;
    }
  </style>
  <div id="box-1" class="main-glyph">
    <div class="left-glyph-asset"></div>
    <input class="comp-1-input" value="Test AAA">
  </div>
  <input class="comp-1-done" value="Test BBB">
  <x-comp1></x-comp1>
  <x-comp1 type="flipper"></x-comp1>
  <script type="application/dart">
import 'dart:async';
import 'dart:html';
import 'package:unittest/unittest.dart';
import 'package:web_ui/web_ui.dart';
import 'package:web_ui/observe/html.dart';


main() {
  useShadowDom = false;

  Timer.run(() {
    // Set of linked style sheet urls.
    var urls = queryAll('link')
        .where((link) => link.attributes['rel'] == 'stylesheet')
        .map((link) => link.attributes['href'])
        .toSet();

    expect(urls.length, 3);
    expect(urls, contains('css_var_test.html.css'));
    expect(urls, contains('css_import.css'));
    expect(urls, contains('resources/assets/css/asset.css'));

    // URI to image in base input directory
    var items = queryAll('#box-1');
    DivElement box1Elem = items[0];
    var image1 = box1Elem.getComputedStyle().backgroundImage;
    expect(image1.endsWith('/input/main.png)'), true);

    // URI to image in directory input/resources
    DivElement box2Elem = box1Elem.children[0];
    var image2 = box2Elem.getComputedStyle().backgroundImage;
    expect(image2.endsWith('input/resources/assets/images/glyph_asset.png)'),
        true);

    window.postMessage('done', '*');
   });
}
  </script>
</body>
</html>
