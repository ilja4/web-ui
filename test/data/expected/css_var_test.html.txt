Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x260
  RenderBlock {HTML} at (0,0) size 800x260
    RenderBody {BODY} at (8,8) size 784x244
      RenderBlock {DIV} at (0,0) size 322x102 [bgcolor=#FF0000] [border: (1px solid #000000)]
        RenderBlock (floating) {DIV} at (11,16) size 14x14
        RenderTextControl {INPUT} at (11,32) size 300x18 [color=#0000FF] [bgcolor=#ADFF2F]
        RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,102) size 784x22
        RenderTextControl {INPUT} at (0,2) size 300x18 [color=#9400D3] [bgcolor=#E0FFFF]
        RenderText {#text} at (0,0) size 0x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,124) size 784x60
        RenderBlock {DIV} at (0,0) size 340x60
          RenderTextControl {INPUT} at (20,20) size 300x20 [color=#FF0000] [bgcolor=#0000FF]
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,184) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,184) size 784x60
        RenderBlock {DIV} at (0,0) size 340x60
          RenderTextControl {INPUT} at (20,20) size 300x20 [color=#FFFF00] [bgcolor=#008000]
          RenderText {#text} at (0,0) size 0x0
      RenderBlock (anonymous) at (0,244) size 784x0
        RenderInline {SPAN} at (0,0) size 0x0
          RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
        RenderText {#text} at (0,0) size 0x0
layer at (19,40) size 300x18
  RenderBlock {DIV} at (0,0) size 300x18
    RenderText {#text} at (0,0) size 80x18
      text run at (0,0) width 80: "Test AAA"
layer at (8,112) size 300x18
  RenderBlock {DIV} at (0,0) size 300x18
    RenderText {#text} at (0,0) size 80x18
      text run at (0,0) width 80: "Test BBB"
layer at (28,153) size 300x18
  RenderBlock {DIV} at (0,1) size 300x18
    RenderText {#text} at (0,0) size 70x18
      text run at (0,0) width 70: "Testing"
layer at (28,213) size 300x18
  RenderBlock {DIV} at (0,1) size 300x18
    RenderText {#text} at (0,0) size 70x18
      text run at (0,0) width 70: "Testing"
